scipy/optimize/_zeros_py.py

```diff
@@ -1396,6 +1398,298 @@ def toms748(f, a, b, args=(), k=1,
+        nfev : int
+            The number of times the function was called to find the root.
+        nit : int
+            The number of iterations of Chandrupatla's algorithm performed.
+        status : int
+            An integer representing the exit status of the algorithm.
+            ``0`` : The algorithm produced a valid bracket.
+            ``-1`` : The bracket expanded to the allowable limits without finding a bracket.
+            ``-2`` : The maximum number of iterations was reached.
+            ``-3`` : A non-finite value was encountered.
+            ``-4`` : Iteration was terminated by `callback`.
+            ``1`` : The algorithm is proceeding normally (in `callback` only).
+            ``2`` : A bracket was found in the opposite search direction (in `callback` only).
+
+        success : bool
+            ``True`` when the algorithm terminated successfully (status ``0``).
+
+    Notes
+    -----
+    This function generalizes an algorithm found in pieces throughout
+    `scipy.stats`. The strategy is to iteratively grow the bracket `(l, r)`
+     until ``func(l) < 0 < func(r)``.
+
+    - If `min` is not provided, the distance between `b` and `l` is iteratively
+      increased by `factor`.
+    - If `min` is provided, the distance between `min` and `l` is iteratively
+      decreased by `factor`. Note that this *increases* the bracket size.
+
+    Growth of the bracket to the right is analogous.
+
+    Growth of the bracket in one direction stops when the endpoint is no longer
+    finite, the function value at the endpoint is no longer finite, or the
+    endpoint reaches its limiting value (`min` or `max`). Iteration terminates
+    when the bracket stops growing in both directions, the bracket surrounds
+    the root, or a root is found (accidentally).
+
+    If multiple brackets are found, only the leftmost one is returned.
+    If roots of the function are found, both `l` and `r` are set to the
+    leftmost root.
+
+    """
+    # Todo:
+    # - find bracket with sign change in specified direction
+    # - Add tolerance
+    # - allow factor < 1?
+
+    callback = None  # works; I just don't want to test it
+    temp = _bracket_root_iv(func, a, b, min, max, factor, args, maxiter)
+    func, a, b, min, max, factor, args, maxiter = temp
+
+    xs = (a, b)
+    temp = _scalar_optimization_initialize(func, xs, args)
+    xs, fs, args, shape, dtype = temp
+    x = np.concatenate(xs)
+    f = np.concatenate(fs)
+    n = len(x) // 2
+
+    x_last = np.concatenate((x[n:], x[:n]))
+    f_last = np.concatenate((f[n:], f[:n]))
+    x0 = x_last
+
+    min = np.broadcast_to(min, shape).astype(dtype, copy=False).ravel()
+    max = np.broadcast_to(max, shape).astype(dtype, copy=False).ravel()
+    limit = np.concatenate((min, max))
+
+    factor = np.broadcast_to(factor, shape).astype(dtype, copy=False).ravel()
+    factor = np.concatenate((factor, factor))
```

Comment:
I am not sure why you are broadcasting this as `work.factor` does not seem to change. Are you considering: 

1. A varying factor
2. Allowing to pass an array

Reply:
I'll have to look, but I thought I allowed for vectorized `factor` - or do I ever change the factor for certain slices? I'm not at a computer. In principle, it doesn't need to be the same for all slices.

Reply:
Yeah, it looks like an array factor is never used, but I wanted to allow for it. I can add a test.

Reply:
Added.